### DISCUSSION TITLE
Inquiring about a safer alternative to dangerouslyForceHydrate

### DISCUSSION THREAD

**Discussion by brain1401** [pmndrs/jotai#2478] [General] [answered]
I'm using the useHydrateAtoms hook in my project and currently relying on the dangerouslyForceHydrate option for rehydration. However, the Jotai docs mention that this option can potentially cause issues with concurrent rendering in React.

I'm a bit hesitant to continue using dangerouslyForceHydrate due to the potential risks. I was wondering if there are any safer alternatives or approaches to rehydrating atoms without using this option.

Is it technically feasible for the Jotai developers to implement a safer way to perform rehydration? If so, could you consider adding this feature in a future release?

If a safer alternative is not currently possible, could you provide some guidance on best practices or precautions to take when using dangerouslyForceHydrate to minimize potential issues?

I'd really appreciate any insights you can share on this topic. My goal is to ensure the stability of my application while still being able to perform the necessary rehydration.

Thanks in advance for your help!

**Comment by dai-shi** [comment, marked as answer]
`dangerouslyForceHydrate` is basically violating React rule. There's not proper way unless React rule changes.
To avoid such a hack, the canonical solution is to use `useEffect`.
See: https://github.com/pmndrs/jotai/discussions/2337#discussioncomment-8092610

**Reply by brain1401** [reply]
I see, thank you for replying!
